File: in_Python/1299 Replace Elements with Greatest Element on Right Side.py
class Solution(object):
    def replaceElements(self, arr):
        """
        :type arr: List[int]
        :rtype: List[int]
        """
        # scan from right to left
        if len(arr) == 1:
            return [-1]
        idx = len(arr)-1
        cur_max = arr[idx]
        while idx >= 0:
            temp = arr[idx-1]
            arr[idx-1] = cur_max
            if cur_max < temp: cur_max = temp
            idx -= 1
        arr[-1] = -1
        return arr

File: in_Java/1299 Replace Elements with Greatest Element on Right Side.java
class Solution {
    public int[] replaceElements(int[] arr) {
        int curmax = arr[arr.length-1];
        arr[arr.length-1] = -1;
        int t;
        for(int i = arr.length-2; i >= 0; i--){
            t = arr[i];
            arr[i] = curmax;
            if(t > curmax) curmax = t;
        }
        return arr;
    }
}